src/KDTree.C

```diff
@@ -71,10 +72,10 @@ void KDTree::nearestNeighbours(SKDCluster pt, int N, SharedKDClusters& result,
   tree->n_nearest(qv, N, vec);
 
   // Sort and transform results
-  this->transformResults(vec, result, filter);
+  this->transformResults(vec, result, std::move(filter));
```

Comment:
can't you make the filter function also a `const&` and then moving shouldn't be done or necessary?